src/core/thread/mle_router.cpp

```diff
@@ -1309,7 +1310,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     case kDeviceStateChild:
         if ((mDeviceMode & ModeTlv::kModeFFD) && (GetActiveRouterCount() < mRouterUpgradeThreshold))
         {
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+            mRouterSelectionJitterTimer.Start(Timer::SecToMsec(otPlatRandomGet() % kRouterSelectionJitter));
```

Comment:
Yes, basically I notice that REED processes Route64TLV as well when receiving the child ID response to determine if it will become an active router or not. However, spec only states that need to wait a delay period when handling advertisement message to prevent REEDs from requesting Router ID simultaneously (advertisement is link-local multicast). So i didn't add this to `Mle::HandleChildIdResponse`. However, it looks like that we need to do this. Now I also understand why each test case of harness has to wait 125s for router upgrade before execution.